### PR TITLE
add a heuristic for deciding when it's safe to enable bullet

### PR DESCRIPTION
**Pull request by sethalves**
- improve the code that decides when to start running physics

### CODE REVIEW

**Comment by sethalves** [comment]
https://app.asana.com/0/26485730907942/108583069039099

**Comment by sethalves** [comment]
to test:
- go to demo and walk into the middle of the table so that your avatar is up against the blocks
- reset the room
- log out
- log in and see if the blocks fall through the table to the floor
- repeat log out / log in a few times

and
- go to "hell"
- once everything is loaded, enable advanced menus and clear cache
- see if you fall through the floor and end up stuck and jittering

**Comment by hifi-gustavo** [comment]
Build console output for commit: e4c91e50648d881b242927f47893d769fd52295d available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

**Comment by sethalves** [comment]
I hate you, Gustavo

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e4c91e50648d881b242927f47893d769fd52295d available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e4c91e50648d881b242927f47893d769fd52295d available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

File: libraries/entities/src/EntityTreeElement.cpp [hunk 1]
```diff
@@ -699,7 +699,7 @@ void EntityTreeElement::getEntities(const glm::vec3& searchPosition, float searc
 
         // if the sphere doesn't intersect with our world frame AABox, we don't need to consider the more complex case
         glm::vec3 penetration;
-        if (success && entityBox.findSpherePenetration(searchPosition, searchRadius, penetration)) {
+        if (!success || entityBox.findSpherePenetration(searchPosition, searchRadius, penetration)) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTreeElement.cpp]: does this mean that all entities that can't get their true AABox will be considered to be "found"? is that really what we want?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 688de6f1004066fca76bf9a63d1db8dd7cdc597b available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

File: libraries/entities/src/EntityTreeElement.cpp (hunk 1 above)
Comment (sethalves) [inline libraries/entities/src/EntityTreeElement.cpp]: I'm not sure.  The test is flawed, either way -- the bounding box isn't the same as the shape of the entity, so you can get false positives even without this change.  I think the rest of the PR stands without this -- I was just trying to get all the entities I could out of getEntities.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 298fe1e1881eded6765c1df1b8ae4258d185c114 available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d6673fb3c08cca453567afad73e1264f94a483e9 available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

**Comment by Nex-Pro** [comment]
Testing this
Did not fall through floor on clearing cache.
Unable to complete test at demo. Despite Chris adding me to whitelist  could not get into demo.
Also default avatar would not load.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ba0549dddd95af62b5e11901bc1d1ab443f87f99 available at: https://gist.github.com/c078f20efa49fa2b52cbe1f87f3a7bf2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7645.dmg

**Pull request merged**